dist/runtime/plugin.mjs

```diff
@@ -0,0 +1,36 @@
+import { defineNuxtPlugin, useRuntimeConfig, useRequestHeaders } from "#app";
+import { reactive } from "vue";
+import generateFlags from "./generateFlags.mjs";
+export default defineNuxtPlugin((nuxtApp) => {
+  const config = useRuntimeConfig();
+  const DEFAULT_USER_AGENT = config.public.device.defaultUserAgent;
+  const REFRESH_ON_RESIZE = config.public.device.refreshOnResize;
+  if (nuxtApp.ssrContext) {
+    const headers = useRequestHeaders();
+    const userAgent2 = headers["user-agent"] || DEFAULT_USER_AGENT;
+    const flags2 = reactive(generateFlags(headers, userAgent2));
+    return {
```

Comment:
@Redemption198 
Been trying to use the refresh on resize, however I was unable due the code stopping from executing because of return within this if statement.

I've tried to modify the code to make it work with this:
```js
  if(nuxtApp.ssrContext){
    const headers = useRequestHeaders();
    const userAgent = headers.userAgent || DEFAULT_USER_AGENT;
    const flags = reactive(generateFlags(headers, userAgent));
    if (REFRESH_ON_RESIZE) {
      window.addEventListener("resize", () => {
        setTimeout(() => {
          const newFlags = generateFlags({}, headers.userAgent || DEFAULT_USER_AGENT);
          Object.entries(newFlags).forEach((entry) => {
            const [key, value] = entry;
            flags[key] = value;
          });
        }, 50);
      });
    }
    return {
      provide: {
        device: flags
      }
    };
  }
```

Which resulted with this error
```
[nuxt] [request error] [unhandled] [500] window is not defined
```

Any thoughts